platform/darwin/src/MGLShapeSource.h

```diff
@@ -210,6 +210,31 @@ MGL_EXPORT
  */
 @property (nonatomic, copy, nullable) NSURL *URL;
 
+#pragma mark Querying the source's features
+
+/**
+ Returns an array of map features for this source, filtered by the given predicate.
+ 
+ Each object in the returned array represents a feature for the
+ current style and provides access to attributes specified by the relevant
+ <a href="https://www.mapbox.com/mapbox-gl-style-spec/#sources">tile sources</a>.
+ The returned array includes features specified in vector and GeoJSON tile
+ sources but does not include anything from raster, image, or video sources.
+ 
+ Features come from tiled vector data or GeoJSON data that is converted to tiles
+ internally, so feature geometries are clipped at tile boundaries and features
+ may appear duplicated across tiles. For example, suppose the specified
+ rectangle intersects with a road that spans the screen. The resulting array
+ includes those parts of the road that lie within the map tiles covering the
+ specified rectangle, even if the road extends into other tiles. The portion of
+ the road within each map tile is included individually.
+ 
+ @param predicate A predicate to filter the returned features.
+ @return An array of objects conforming to the `MGLFeature` protocol that
+ represent features in the sources used by the current style.
+ */
+- (NS_ARRAY_OF(id <MGLFeature>) *)featuresWithPredicate:(nullable NSPredicate *)predicate NS_SWIFT_NAME(features(predicate:));
```

Comment:
Call this method `-featuresMatchingPredicate:` and remove the `NS_SWIFT_NAME` attribute. Swift will automatically bridge it as `features(matching:)`.

The convention in Swift 3 is for argument labels to be prepositions or participles like “matching” instead of prepositional objects like “predicate”. `matching:` would be more intuitive argument label than `with:` in this case. (It’d also be consistent with methods like `-[CNContactStore groupsMatchingPredicate:error:]`, `-[EKEventStore eventsMatchingPredicate:]`, and `-[XCUIElementQuery matchingPredicate:]` in standard Apple libraries.)